File: golang/two_sum_ii_input_array_is_sorted.go
package main

// Space : O(n)
// Time  : O(n)

func twoSum(numbers []int, target int) []int {
    n := len(numbers)
    hashmap := make(map[int]int)
    
    for i := 0; i < n; i++ {
        hashmap[numbers[i]] = i+1
    }
    
    for i := 0; i < n; i++ {
        if _, ok := hashmap[target - numbers[i]]; ok {
            return []int{i+1, hashmap[target - numbers[i]]} 
        }
    }
    
    return []int{}
}

File: python3/two_sum_ii_input_array_is_sorted.py
# Space : O(n)
# Time  : O(n)

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        hashmap = {numbers[i]: (i+1) for i in range(n)}

        for i in range(n):
            if target - numbers[i] in hashmap:
                return [i+1, hashmap[target - numbers[i]]]

        return []
